### PR TITLE
Add Dockerfile for indexer-balances package

### PR DESCRIPTION
**Pull request by ecp88**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+FROM rust:1.62 AS builder
+WORKDIR /tmp/
+COPY Cargo.toml Cargo.lock ./
+COPY src/lib.rs src/lib.rs
```
Comment (frol) [inline Dockerfile]: Any change in `lib.rs` will invalidate incremental builds, so I recommend to follow the pattern we have in Enhacned API:

https://github.com/near/near-enhanced-api-server/blob/main/Dockerfile#L5

```Dockerfile
RUN mkdir src && echo 'fn main() {}' > src/main.rs && cargo build --release && rm -r src
```

Adjusting it to lib-crate:

```Dockerfile
RUN mkdir src && touch src/lib.rs && cargo build --release && rm -r src
```

**Review by frol — APPROVED** [review]
LGMT. Please, address the comment

File: Dockerfile (hunk 1 above)
Comment (ecp88) [reply Dockerfile]: Updated with your line. But as is we get :
```
=> ERROR [builder 5/7] RUN cargo build --release && rm -r src                                                                                                                                            0.6s
------                                                                                                                                                                                                         
 > [builder 5/7] RUN cargo build --release && rm -r src:                                                                                                                                                       
#12 0.597 error: failed to parse manifest at `/tmp/Cargo.toml`                                                                                                                                                 
#12 0.597 
#12 0.597 Caused by:
#12 0.597   can't find library `indexer-balances`, rename file to `src/lib.rs` or specify lib.path
------
```
So I added the `COPY src/lib.rs src/lib.rs` after your command to successfully build.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
+FROM rust:1.62 AS builder
+WORKDIR /tmp/
+COPY Cargo.toml Cargo.lock ./
+RUN mkdir src && touch src/lib.rs && cargo build --release && rm -r src
+
+# this build step will cache your dependencies
+COPY src/lib.rs src/lib.rs
+RUN cargo build --release && rm -r src
```
Comment (frol) [inline Dockerfile]:
```suggestion

# this build step will cache your dependencies
COPY Cargo.toml Cargo.lock ./
RUN mkdir src && touch src/lib.rs && cargo build --release && rm -r src
```

File: Dockerfile (hunk 1 above)
Comment (frol) [reply Dockerfile]: See my comment below

File: Dockerfile (hunk 2 above)
Comment (ecp88) [reply Dockerfile]: Done. Thanks!

File: Dockerfile (hunk 1 above)
Comment (ecp88) [reply Dockerfile]: Updated. Thanks!

**Pull request merged**